packages/flutter_tools/lib/runner.dart

```diff
@@ -164,6 +164,10 @@ Future<int> _handleToolError(
   String Function() getFlutterVersion,
   ShutdownHooks shutdownHooks,
 ) async {
+
+  bool isDueToGitMissing(ProcessException error) => error.message.contains('git') &&
+      !globals.processManager.canRun('git');
```

Comment:
what if .canRun() throws?

Reply:
Great question.

I think wrapping this call in `try` is a good idea.

Reply:
been bitten by this one before, unfortunately

Reply:
addressed in [wrap .canRun for git with try-catch](https://github.com/flutter/flutter/pull/154445/commits/5319638b6b51af56cc59d51622516f9df6b549cb)